Assets/Scripts/New/MainMenu.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
```

Comment:
If the UI logic need these many serialized references, we will need to take a different approach than a simple UI Manager. We will have to implement MVC for each UI Screen. I have done this for the state machine project and the command pattern project. Sync with me to understand & implement this logic.